src/EditorFeatures/CSharpTest/UseLocalFunction/UseLocalFunctionTests.cs

```diff
@@ -1993,6 +1993,76 @@ void Goo()
         var buildCancelled = false;
         void onUpdateSolutionCancel(int a) { buildCancelled = true; }
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(23149, "https://github.com/dotnet/roslyn/issues/23149")]
+        public async Task TestNotAvaliableIfTypeParameterChanged1()
+        {
+            await TestMissingAsync(
+@"using System;
+
+class Enclosing<T>
+{
+    delegate T MyDelegate(T t);
+    static void Callee(MyDelegate d) => d(default);
+
+    public class Class<T>
+    {
+        public void Caller()
+        {
+            MyDelegate [||]local = x => x;
+            Callee(local);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(23149, "https://github.com/dotnet/roslyn/issues/23149")]
+        public async Task TestNotAvaliableIfTypeParameterChanged2()
+        {
+            await TestMissingAsync(
+@"using System;
+
+class Enclosing<T>
+{
+    delegate T MyDelegate(T t);
+    static void Callee(MyDelegate d) => d(default);
+
+    public class Foo<T>
+    {
+        public class Class
```

Comment:
Let's add a similar test where the nested type declares some additional type parameters of its own: `class Class<U> ...`

Reply:
➡️ This is the test demonstrating the following issue https://github.com/dotnet/roslyn/pull/23465#discussion_r191032625